src/main/scala/centaur/test/Test.scala

```diff
@@ -142,7 +182,66 @@ object Operations {
         }
       }
 
-      override def run: Try[Unit] = workflowLengthFutureCompletion(() => Future { doPerform() })
+      override def run: Try[Unit] = sendReceiveFutureCompletion(() => Future { doPerform() })
+    }
+  }
+
+  def validatePAPIAborted(jobId: String, workflow: SubmittedWorkflow): Test[Unit] = {
+    import scala.concurrent.duration._
+
+    def eventually(startTime: OffsetDateTime, timeout: FiniteDuration)(f: => Try[Unit]): Try[Unit] = {
+      f match {
+        case Failure(_) if OffsetDateTime.now().isBefore(startTime.plusSeconds(timeout.toSeconds)) =>
```

Comment:
ToL: `now` instead of `now()`?

Reply:
I think `now()` is right here because it's not a pure function

Reply:
ooh right. it's not side effecting but it's sure as heck not referentially transparent. 😄